src/service.rs

```diff
@@ -0,0 +1,273 @@
+                        None => "Unknown".to_string(),
+                    };
+
+                    let state_width = state_string.len() + 2;
+                    let name_width = name_string.len() + 2;
+
+                    println!("{0: <state_width$} {1: <name_width$} Display Name", "State", "Name", state_width = state_width, name_width = name_width);
+                    println!("{0:-<state_width$} {1:-<name_width$} -------------", "-", "-", state_width = state_width, name_width = name_width);
+                    match maybe_config {
+                        Some(config) => {
+                            println!("{0: <state_width$} {1: <name_width$} {2:#?}", state_string, name_string, config.display_name, state_width = state_width, name_width = name_width);
+                            exitcode::OK
+                        }
+                        _ => exitcode::SOFTWARE
+                    }
+
+                },
+                Err(error) => {
+                    error!(message = "Error opening service.", %error);
+                    exitcode::SOFTWARE
+                }
+            }
+        },
+        Err(error) => {
+            error!(message = "Error connecting to service manager.", %error);
+            exitcode::SOFTWARE
+        }
+    };
+
+    code
+}
+
+#[cfg(windows)]
+fn control_service(service: &ServiceInfo, action: ControlAction) -> exitcode::ExitCode {
+    use crate::vector_windows;
+
+    if action == ControlAction::Status {
+        return print_status(&service);
+    }
+
+   let service_definition =  vector_windows::service_control::ServiceDefinition {
+       name: service.name.clone(),
+       display_name: service.display_name.clone(),
+       description: service.description.clone(),
+       executable_path: service.executable_path.clone(),
+       launch_arguments: service.launch_arguments.clone(),
+   };
+
+    let control_action = match action {
+        ControlAction::Install => Some(vector_windows::service_control::ControlAction::Install),
+        ControlAction::Uninstall => Some(vector_windows::service_control::ControlAction::Uninstall),
+        ControlAction::Start => Some(vector_windows::service_control::ControlAction::Start),
+        ControlAction::Stop => Some(vector_windows::service_control::ControlAction::Stop),
+        _ => None,
+    };
+
+    let exit_code = match control_action {
+        Some(a) => {
+            match vector_windows::service_control::control(&service_definition, a) {
+                Ok(()) => exitcode::OK,
+                Err(error) => {
+                    error!(message = "Error controlling service.", %error);
+                    exitcode::SOFTWARE
+                },
+            }
+        }
+        None => exitcode::USAGE,
+    };
+
+    exit_code
+}
+
+#[cfg(unix)]
+fn control_service(service: &ServiceInfo, action: ControlAction) -> exitcode::ExitCode {
+    error!("Service commands are currently not supported on this platform.");
```

Comment:
I made the choice of making the `service` subcommand available on all platforms, but the only implementation is for Windows right now.

This gives the opportunity  to add other platforms implementation later on.

Reply:
I think for now it's good to make it only appear on windows. We can make it appear later for other OS. :)

Reply:
Ok so I'll make the sub command only available on Windows, then.

Reply:
Thank you! :)